game/scripts/vscripts/modifiers/modifier_sonic_fly.lua

```diff
@@ -0,0 +1,24 @@
+modifier_sonic_fly = class(ModifierBaseClass)
+
+function modifier_sonic_fly:DeclareFunctions()
+  return {
+      MODIFIER_PROPERTY_MOVESPEED_BONUS_CONSTANT,
+      MODIFIER_PROPERTY_BONUS_VISION_PERCENTAGE,
+      --MODIFIER_PROPERTY_STATUS_RESISTANCE
+  }
```

Comment:
I think you should leave the status resistance code in. Though I'm unsure if it will support negative values even when Valve does fix it.